execution_engine_testing/tests/src/test/explorer/faucet.rs

```diff
@@ -1,77 +1,1210 @@
+                    ARG_TIME_INTERVAL => 10_000u64,
+                    ARG_DISTRIBUTIONS_PER_INTERVAL => 2u64
+                },
+            )
+            .with_empty_payment_bytes(runtime_args! {ARG_AMOUNT => *DEFAULT_PAYMENT})
+            .with_deploy_hash([3; 32])
+            .build();
+
+        ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+    };
+
+    builder
+        .exec(installer_set_variable_request)
+        .expect_success()
+        .commit();
+
+    let available_amount = builder
+        .query(
+            None,
+            faucet_contract_hash.into(),
+            &[AVAILABLE_AMOUNT_NAMED_KEY.to_string()],
+        )
+        .expect("failed to find available amount named key")
+        .as_cl_value()
+        .cloned()
+        .expect("failed to convert to cl value")
+        .into_t::<U512>()
+        .expect("failed to convert into U512");
+
+    assert_eq!(available_amount, U512::from(500_000_000u64));
+
+    // This would only allow other callers to fund twice in this interval,
+    // but the installer can fund as many times as they want.
+    for num in 0..3 {
+        let faucet_call_by_installer = {
+            let deploy_item = DeployItemBuilder::new()
+                .with_address(installer_account)
+                .with_authorization_keys(&[installer_account])
+                .with_stored_session_named_key(
+                    FAUCET_CONTRACT_NAMED_KEY,
+                    ENTRY_POINT_FAUCET,
+                    runtime_args! {ARG_TARGET => user_account, ARG_ID => <Option<u64>>::None},
+                )
+                .with_empty_payment_bytes(runtime_args! {ARG_AMOUNT => *DEFAULT_PAYMENT})
+                .with_deploy_hash([num + 4; 32])
+                .build();
+
+            ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+        };
+
+        builder
+            .exec(faucet_call_by_installer)
+            .expect_success()
+            .commit();
+    }
+
+    let faucet_contract = builder
+        .get_contract(faucet_contract_hash)
+        .expect("failed to find faucet contract");
+
+    let faucet_purse = faucet_contract
+        .named_keys()
+        .get(FAUCET_PURSE_NAMED_KEY)
+        .cloned()
+        .and_then(Key::into_uref)
+        .expect("failed to find faucet purse");
+
+    let faucet_purse_balance = builder.get_purse_balance(faucet_purse);
+    assert_eq!(
+        faucet_purse_balance,
+        U512::from(200_000_000_000u64 - 750_000_000u64)
     );
+
+    // check the balance of the user's main purse
+    let user_main_purse_balance_after =
+        builder.get_purse_balance(builder.get_expected_account(user_account).main_purse());
+    //                 available_amount
+    //                 ________________
+    //                /
+    //               /  distributions_per_interval
+    //              /   __________________________
+    //  ___________/  _/
+    // (500_000_000 / 2 = 250_000_000) * 3 = 750_000_000
```

Comment:
Given that this is the very end of the last test, it's a little odd to find the first fairly explicit comment here :)

Probably worth just removing this I think.